language/move-command-line-common/src/character_sets.rs

```diff
@@ -15,10 +15,12 @@ pub fn is_permitted_printable_char(c: char) -> bool {
 
 /// Determine if a character is a permitted newline character.
 ///
-/// The only permitted newline character is \n. All others are invalid.
+/// The only permitted newline character is \n or \r\n. All others are invalid.
 pub fn is_permitted_newline_char(c: char) -> bool {
     let x = c as u32;
-    x == 0x0A
+    let is_cr = x == 0x0D; // \r
```

Comment:
If I understand correctly this would allow a standalone `\r`. However, we only want to allow exactly `\r\n`, and no standalone CR because of the risk of source 'hiding' things when printed, 

Can you modify this PR to only allow CR if it is followed by a NL?

Reply:
OK，I try it.

Reply:
@wrwg I have modified this PR to only allow CR if it is followed by a LF.